source/slang/slang-parser.cpp

```diff
@@ -3925,7 +3953,9 @@ namespace Slang
         {
             FillPosition(blockStatement);
         }
-        while (!AdvanceIfMatch(this, TokenType::RBrace))
+
+        Token closingBraceToken;
+        while (!AdvanceIfMatch(this, TokenType::RBrace, &closingBraceToken))
```

Comment:
If we do a follow-up change where we add a `SourceRange` to each and every AST node, we might want to consider making some of the tracking of source ranges in the parser more systematic/automatic. Having to capture a token from a subroutine here so that we can poke it into the `blockStatement` feels like the kind of thing we want to make more automatic if we are going to apply it broadly throughout the compiler.

(No concrete request here; just making notes for when we move on to future stuff we've talked about like documentation comments)